python-threatexchange/threatexchange/fetcher/fetch_api.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+The fetcher is the component that talks to external APIs to get and put signals
+
+@see SignalExchangeAPI
+"""
+
+
+import typing as t
+
+from threatexchange.collab_config import CollaborationConfig
+from threatexchange.signal_type.signal_base import SignalType
+from . import fetch_state_base as state
+
+
+class SignalExchangeAPI:
+    """
+    An external APIs to get and maybe put signals
+
+    Fetchers ideally can checkpoint their progress, so that they can tail
+    updates from the server.
+
+    Is assumed that fetched signals have some metadata attached to them,
+    which is unique to that API. Additionally, it a assumed that there
+    might be multiple contributors to signals inside of an API.
+
+    The Fetcher can assume it is stateful, and
+    """
+
+    @classmethod
+    def name(cls) -> str:
+        """
+        A string name unique to all SignalExchangeAPIs
+
+        This will get stored for lookup from a map of supported APIs
+        keyed str: API
+        """
+        return cls.__name__
+
+    def resolve_owner(self, id: int) -> str:
+        """
+        Convert an owner ID into a human readable name (if available)
+
+        If empty is returned, its assumed that the collaboration name itself
+        is good enough to describe this content.
+        """
+        return ""
+
+    def get_own_owner_id(self) -> int:
+        """
+        Return the owner ID of this caller. Opinions with that ID are "ours"
+        """
+        return -1
```

Comment:
nit: why` -1` instead of `raise` or `None`

Reply:
Picking an implementation that will probably do the right thing if unimplemented. Might have to change eventually. Overall goal was to leave the minimum number of methods to get something working. Having a self owner ID isn't strictly needed to fetch a first round of hashes, so provided a default implementation.